1-org/modules/centralized-logging/main.tf

```diff
@@ -0,0 +1,205 @@
+  //  source  = "terraform-google-modules/log-export/google//modules/logbucket"
+  //  version = "~> 7.4.2"
+
+  source = "github.com/terraform-google-modules/terraform-google-log-export//modules/logbucket"
+
+  count = local.create_destination && var.logging_target_type == "logbucket" ? 1 : 0
+
+  project_id                    = var.logging_destination_project_id
+  name                          = local.logging_target_name
+  log_sink_writer_identity      = module.log_export[local.key_first_resource].writer_identity
+  location                      = var.logging_location
+  retention_days                = var.retention_days
+  grant_write_permission_on_bkt = false
+}
+
+#-------------------------------------------#
+# Log Bucket Service account IAM membership #
+#-------------------------------------------#
+resource "google_project_iam_member" "logbucket_sink_member" {
+  for_each = var.logging_target_type == "logbucket" ? local.logbucket_sink_member : {}
+
+  project = var.logging_destination_project_id
+  role    = "roles/logging.bucketWriter"
+  member  = module.log_export[each.key].writer_identity
+}
+
+#-----------------------#
+# Send logs to BigQuery #
+#-----------------------#
+module "destination_bigquery" {
+  source  = "terraform-google-modules/log-export/google//modules/bigquery"
+  version = "~> 7.3.0"
+
+  count = local.create_destination && var.logging_target_type == "bigquery" ? 1 : 0
+
+  project_id                 = var.logging_destination_project_id
+  dataset_name               = replace(local.logging_target_name, "-", "_")
+  log_sink_writer_identity   = module.log_export[local.key_first_resource].writer_identity
+  labels                     = var.labels
+  description                = var.dataset_description
+  kms_key_name               = var.kms_key_name
+  expiration_days            = var.expiration_days
+  delete_contents_on_destroy = var.delete_contents_on_destroy
+}
+
+#-----------------------------------------#
+# Bigquery Service account IAM membership #
+#-----------------------------------------#
+resource "google_project_iam_member" "bigquery_sink_member" {
+  for_each = var.logging_target_type == "bigquery" ? var.resources : {}
+
+  project = var.logging_destination_project_id
+  role    = "roles/bigquery.dataEditor"
+  member  = module.log_export[each.key].writer_identity
+}
+
+
+#----------------------#
+# Send logs to Storage #
+#----------------------#
+module "destination_storage" {
+  source  = "terraform-google-modules/log-export/google//modules/storage"
+  version = "~> 7.3.0"
+
+  count = local.create_destination && var.logging_target_type == "storage" ? 1 : 0
+
+  project_id                  = var.logging_destination_project_id
+  storage_bucket_name         = local.logging_target_name
+  log_sink_writer_identity    = module.log_export[local.key_first_resource].writer_identity
+  kms_key_name                = var.kms_key_name
+  uniform_bucket_level_access = var.uniform_bucket_level_access
```

Comment:
We can hardcode most of these like uniform_bucket_level_access to true in the module.

Reply:
Done